File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        s_count = {}
        t_count = {}

        for c in s:
            s_count[c] = 1 + s_count.get(c, 0)

        for c in t:
            t_count[c] = 1 + t_count.get(c, 0)
        
        return t_count == s_count

File: 0242-valid-anagram/0242-valid-anagram.rs
use std::collections::HashMap;
impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.len() != t.len(){
            return false;
        }
        let mut count = HashMap::new();

        for (s_char, t_char) in s.chars().zip(t.chars()){
            *count.entry(s_char).or_insert(0) += 1;
            *count.entry(t_char).or_insert(0) -= 1;
        } 

        return count.values().all(|&v| v == 0);
    }
}